src/pudl/extract/ferc1.py

```diff
@@ -860,6 +862,54 @@ def extract_xbrl(
     return ferc1_raw_dfs
 
 
+def steam_xbrl(
+    ferc1_engine: sa.engine.Engine, ferc1_settings: Ferc1XbrlSettings
+) -> pd.DataFrame:
+    """Creates a of DataFrame of steam_electric_generating_plant_statistics_large_plants_402_duration.
+
+    Args:
+        ferc1_engine: An SQL Alchemy connection engine for the FERC Form 1 database.
+        ferc1_settings: Object containing validated settings relevant to FERC Form 1.
+    """
+    ferc1_meta = get_ferc1_meta(ferc1_engine)
```

Comment:
I would love if someone... maybe even you zane lol... to take on the refactoring of the extract step. it's pretty clear on a quick look that it can be DRASTICALLY consolidated overall, but i'd like to stay focused on the transform step